index.php

```diff
@@ -819,24 +922,125 @@ public function currentStep() {
 	 * @throws Exception
 	 */
 	public function rollbackChanges($step) {
+		$this->silentLog('[info] rollbackChanges("' . $step . '")');
+
 		$updaterDir = $this->getDataDirectoryLocation() . '/updater-'.$this->getConfigOption('instanceid');
 		if(file_exists($updaterDir . '/.step')) {
+			$this->silentLog('[info] unlink .step');
 			$state = unlink($updaterDir . '/.step');
 			if ($state === false) {
 				throw new \Exception('Could not delete .step');
 			}
 		}
 
 		if($step >= 7) {
+			$this->silentLog('[info] rollbackChanges - step >= 7');
 			// TODO: If it fails after step 7: Rollback
 		}
+		$this->silentLog('[info] end of  rollbackChanges()');
+	}
+
+	/**
+	 * Logs an exception with current datetime prepended to updater.log
+	 *
+	 * @param Exception $e
+	 * @throws LogException
+	 */
+	public function logException(\Exception $e) {
+		$message = '[error] ';
+
+		$message .= 'Exception: ' . get_class($e) . PHP_EOL;
+		$message .= 'Message: ' . $e->getMessage() . PHP_EOL;
+		$message .= 'Code:' . $e->getCode() . PHP_EOL;
+		$message .= 'Trace:' . PHP_EOL . $e->getTraceAsString() . PHP_EOL;
+		$message .= 'File:' . $e->getFile() . PHP_EOL;
+		$message .= 'Line:' . $e->getLine() . PHP_EOL;
+		if($e instanceof UpdateException) {
+			$message .= 'Data:' . PHP_EOL . print_r($e->getData(), true) . PHP_EOL;
+		}
+		$this->log($message);
+	}
+
+	/**
+	 * Logs a message with current datetime prepended to updater.log
+	 *
+	 * @param string $message
+	 * @throws LogException
+	 */
+	public function log($message) {
+		$updaterLogPath = $this->getDataDirectoryLocation() . '/updater.log';
+
+		$fh = fopen($updaterLogPath, 'a');
+		if($fh === false) {
+			throw new \LogException('Could not open updater.log');
+		}
+
+		if($this->requestID === null) {
+			$characters = '0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+			$charactersLength = strlen($characters);
+			$randomString = '';
+			for ($i = 0; $i < 10; $i++) {
+				$randomString .= $characters[rand(0, $charactersLength - 1)];
```

Comment:
I don't want people complaining about that. Can we just use http://php.net/manual/en/function.random-int.php? 

Also you might want to add `$_SERVER['UNIQUE_ID']` which is used by http://httpd.apache.org/docs/current/mod/mod_unique_id.html